thymeleaf-spring6/src/main/java/org/thymeleaf/spring5/context/webmvc/SpringWebMvcThymeleafRequestContext.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * =============================================================================
+ *
+ *   Copyright (c) 2011-2018, The THYMELEAF team (http://www.thymeleaf.org)
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ * =============================================================================
+ */
+package org.thymeleaf.spring5.context.webmvc;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TimeZone;
+
+import jakarta.servlet.http.HttpServletRequest;
```

Comment:
See how, in the whole code submitted here for the new `thymeleaf-spring6` module, there are only two instances of imports on `jakarta.*` classes. This kind of makes me think that the effect on code complexity of performing the `org.thymeleaf.web.servlet.*` generalisation I mentioned for the core module might not be _that_ bad…

Reply:
Are you thinking we would try to have the same code base between spring 5 and 6? Having the same code base for Spring 5 & 6 seems to deviate from the pattern that is currently established and I think it would likely end up making things difficult to maintain.

Reply:
No no, I didn't mean that. I believe these should still be different modules. I was just reflecting on the fact that, after all, use of _native_ `javax.servlet` characteristics is not very extense in these modules, so the abstraction of these entities would not really bring a lot of changes to them.